src/install/install.zig

```diff
@@ -1265,34 +1268,74 @@ const PackageInstall = struct {
     }
 
     pub fn installFromLink(this: *PackageInstall, skip_delete: bool) Result {
-
+        const dest_path = this.destination_dir_subpath;
         // If this fails, we don't care.
         // we'll catch it the next error
-        if (!skip_delete and !strings.eqlComptime(this.destination_dir_subpath, ".")) this.uninstall() catch {};
+        if (!skip_delete and !strings.eqlComptime(dest_path, ".")) this.uninstall() catch {};
 
         // cache_dir_subpath in here is actually the full path to the symlink pointing to the linked package
         const symlinked_path = this.cache_dir_subpath;
 
-        std.os.symlinkatZ(symlinked_path, this.destination_dir.dir.fd, this.destination_dir_subpath) catch |err| {
-            return Result{
+        const subdir = std.fs.path.dirname(dest_path);
+        var dest_dir = if (subdir) |dir| brk: {
+            break :brk this.destination_dir.dir.makeOpenPath(dir, .{}) catch |err| return Result{
                 .fail = .{
                     .err = err,
                     .step = .linking,
                 },
             };
+        } else this.destination_dir.dir;
+        defer {
+            if (subdir != null) dest_dir.close();
+        }
+
+        var dest_full_path = dest_dir.realpathAlloc(this.allocator, ".") catch |err| return Result{
```

Comment:
don't use realpath, use `getFdPath`. We don't need to allocate memory here

https://github.com/oven-sh/bun/blob/520ecc21ac4d46eb102215f39b5e3565dcf2eef4/src/fs.zig#L112

Reply:
Section revamped to use stack instead of allocations in https://github.com/oven-sh/bun/pull/1764/commits/c933a28c2ebb02eda1c342d6a6f346bea53c1c1c